d2/src/test/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerTest.java

```diff
@@ -531,6 +533,182 @@ public void done()
     }
   }
 
+  @Test
+  public void testGetClientWithOptimizedAffinityRouting() throws Exception
+  {
+    Map<String, LoadBalancerStrategyFactory<? extends LoadBalancerStrategy>> loadBalancerStrategyFactories =
+        new HashMap<>();
+    Map<String, TransportClientFactory> clientFactories = new HashMap<>();
+    List<String> prioritizedSchemes = new ArrayList<>();
+
+    MockStore<ServiceProperties> serviceRegistry = new MockStore<>();
+    MockStore<ClusterProperties> clusterRegistry = new MockStore<>();
+    MockStore<UriProperties> uriRegistry = new MockStore<>();
+
+    ScheduledExecutorService executorService = new SynchronousExecutorService();
+
+    loadBalancerStrategyFactories.put("degrader", new DegraderLoadBalancerStrategyFactoryV3());
+    clientFactories.put(PropertyKeys.HTTP_SCHEME, new DoNothingClientFactory());
+
+    SimpleLoadBalancerState state =
+        new SimpleLoadBalancerState(executorService,
+            uriRegistry,
+            clusterRegistry,
+            serviceRegistry,
+            clientFactories,
+            loadBalancerStrategyFactories);
+
+    SimpleLoadBalancer loadBalancer =
+        new SimpleLoadBalancer(state, 5, TimeUnit.SECONDS, _d2Executor);
+
+    FutureCallback<None> balancerCallback = new FutureCallback<>();
+    loadBalancer.start(balancerCallback);
+    balancerCallback.get();
```

Comment:
can this set-up be done with the setupLoadBalancer method at line 172? same for the other test.

Reply:
sure. done